src/runtime_strings_extra.js

```diff
@@ -64,7 +67,7 @@ function UTF16ToString(ptr) {
     var str = '';
     while (1) {
       var codeUnit = {{{ makeGetValue('ptr', 'i*2', 'i16') }}};
-      if (codeUnit == 0) return str;
+      if (codeUnit == 0 || i == maxBytesToRead / 2) return str;
```

Comment:
Should this not be `i == maxIdx`? Having a division in the hot per-characterr loop would be bad for performance.

Reply:
Also note the inconsistency in implementation between UTF16ToString() and UTF32ToString: this retains a `while(1)` whereas UTF32ToString switched to a `while (!(i >= maxBytesToRead / 4)) {`. This variant is better for code size and less branches (termination test is preserved in one if, whereas it is in two parts in UTF32ToString), let's use UTF16ToString() format in both?

Reply:
+1 for a followup to make the code more consistent.